### PR TITLE
fix: Update for default WAF-alignment - `avm/res/cache/redis`

### LINKED ISSUE #1504: [WAF - Reliability Review] - `avm/res/cache/redis`
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Issue Type?

Feature Request

### Module Name

avm/res/cache/redis

### (Optional) Module Name if not listed above

_No response_

### (Optional) Module Version

_No response_

### Description

Hello team,

The AVM core team are performing a review of all AVM modules to ensure they are aligned to the WAF reliability pillar by default, and in the future they will be checking against other WAF pillars also. This is a requirement we have before we can go `v1.0.0` for any AVM module.

Links to AVM specs:

- https://azure.github.io/Azure-Verified-Modules/specs/shared/#id-sfr2---category-composition---waf-aligned
- https://azure.github.io/Azure-Verified-Modules/specs/shared/#id-sfr5---category-composition---availability-zones

Links to WAF Reliability, APRL & Product Docs recommendations used for analysis:

- https://learn.microsoft.com/azure/well-architected/reliability/checklist
- https://learn.microsoft.com/en-us/azure/azure-cache-for-redis/cache-how-to-zone-redundancy?toc=%2Fazure%2Freliability%2Ftoc.json&bc=%2Fazure%2Freliability%2Fbreadcrumb%2Ftoc.json
- https://learn.microsoft.com/en-us/azure/well-architected/service-guides/azure-cache-redis/reliability
- https://azure.github.io/Azure-Proactive-Resiliency-Library/services/database/redis-cache/
- https://azure.github.io/PSRule.Rules.Azure/en/rules/Azure.Redis.AvailabilityZone/#configure-with-bicep

Whilst reviewing this module the following issues have been found that are non-compliant with the AVM specs that need to be resolved in a new version release:

- [Change the parameter of `zoneRedundant` to default to `true`](https://github.com/Azure/bicep-registry-modules/blob/main/avm/res/cache/redis/main.bicep#L94) 
- [Change the parameter of `skuName` to default to `Premium`](https://github.com/Azure/bicep-registry-modules/blob/main/avm/res/cache/redis/main.bicep#L82)
- [Change the parameter of `zones` to default to `[1,2,3]`](https://github.com/Azure/bicep-registry-modules/blob/main/avm/res/cache/redis/main.bicep#L97)
    - May be able to remove the var for AZ logic https://github.com/Azure/bicep-registry-modules/blob/main/avm/res/cache/redis/main.bicep#L108-L110
- [Change the parameter of `replicasPerMaster` to default to `3`](https://github.com/Azure/bicep-registry-modules/blob/main/avm/res/cache/redis/main.bicep#L54)

Please make these changes in a PR to the parameter defaults so a new minor version can be published.

Thanks

The AVM Core Team

### (Optional) Correlation Id

_No response_

### PR DESCRIPTION
**Pull request by hundredacres**
## Description

Updating default to be WAF-aligned. Fixes #1504

## Pipeline Reference

| Pipeline |
| -------- |
| [avm.res.cache.redis](https://github.com/hundredacres/bicep-registry-modules/actions/workflows/avm.res.cache.redis.yml)         |

## Type of Change

- [ ] Update to CI Environment or utlities (Non-module effecting changes)
- [ ] Azure Verified Module updates:
  - [X] Bugfix containing backwards compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [X] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

## Checklist

- [X] I'm sure there are no other open Pull Requests for the same update/change
- [X] I have run `Set-AVMModule` locally to generate the supporting module files.
- [X] My corresponding pipelines / checks run clean and green without any errors or warnings

### CODE REVIEW

File: .github/workflows/avm.ptn.network.hub-networking.yml [hunk 1]
```diff
@@ -0,0 +1,87 @@
+# >>> UPDATE to for example "avm.res.key-vault.vault"
```
Comment (AlexanderSehr) [inline .github/workflows/avm.ptn.network.hub-networking.yml:1]:
```suggestion
```

File: .github/workflows/avm.ptn.network.hub-networking.yml
Comment (AlexanderSehr) [inline .github/workflows/avm.ptn.network.hub-networking.yml:1]: I guess this sneaked its way in from the main branch

File: avm/res/cache/redis/main.bicep [hunk 2]
```diff
@@ -94,7 +94,7 @@ param tenantSettings object = {}
 param zoneRedundant bool = true
 
 @description('Optional. If the zoneRedundant parameter is true, replicas will be provisioned in the availability zones specified here. Otherwise, the service will choose where replicas are deployed.')
-param zones array = []
+param zones array = [1,2,3]
```
Comment (AlexanderSehr) [inline avm/res/cache/redis/main.bicep:97]:
```suggestion
param zones int[] = [1,2,3]
```

Comment (AlexanderSehr) [reply avm/res/cache/redis/main.bicep:97]: Also please note a merge conflict sneaked its way in

File: .github/workflows/avm.ptn.network.hub-networking.yml (hunk 1 above)
Comment (hundredacres) [reply .github/workflows/avm.ptn.network.hub-networking.yml]: Removed

File: .github/workflows/avm.ptn.network.hub-networking.yml
Comment (hundredacres) [reply .github/workflows/avm.ptn.network.hub-networking.yml:1]: Yeah, this is a result of me adding testing for the pattern module. Should I remove it?

File: avm/res/cache/redis/main.bicep (hunk 2 above)
Comment (hundredacres) [reply avm/res/cache/redis/main.bicep]: Fixed.

File: .github/workflows/avm.ptn.network.hub-networking.yml
Comment (AlexanderSehr) [reply .github/workflows/avm.ptn.network.hub-networking.yml:1]: Please do, yes. It's anyways only in this branch. You don't have to remove it from your `main` as you're correct that you need it there for testing.

Comment (hundredacres) [reply .github/workflows/avm.ptn.network.hub-networking.yml:1]: Done.

**Review by AlexanderSehr — APPROVED** [review]

**Pull request merged**